src/v/cloud_storage/topic_manifest.cc

```diff
@@ -35,74 +171,140 @@ topic_manifest::topic_manifest(
 topic_manifest::topic_manifest()
   : _topic_config(std::nullopt) {}
 
+void topic_manifest::update(const topic_manifest_handler& handler) {
+    if (handler._version != topic_manifest_version) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format,
+          "topic manifest version {} is not supported",
+          handler._version));
+    }
+    _rev = handler._revision_id.value();
+
+    if (!handler._version) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format, "Missing _version value in parsed topic manifest"));
+    }
+    if (!handler._namespace) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format, "Missing _namespace value in parsed topic manifest"));
+    }
+    if (!handler._topic) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format, "Missing _topic value in parsed topic manifest"));
+    }
+    if (!handler._partition_count) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format,
+          "Missing _partition_count value in parsed topic manifest"));
+    }
+    if (!handler._replication_factor) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format,
+          "Missing _replication_factor value in parsed topic manifest"));
+    }
+    if (!handler._revision_id) {
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format, "Missing _revision_id value in parsed topic manifest"));
+    }
+
+    _topic_config = cluster::topic_configuration(
+      handler._namespace.value(),
+      handler._topic.value(),
+      handler._partition_count.value(),
+      handler._replication_factor.value());
+    _topic_config->properties = handler._properties;
+
+    if (handler.compaction_strategy_sv) {
+        try {
+            _topic_config->properties.compaction_strategy
+              = boost::lexical_cast<model::compaction_strategy>(
+                handler.compaction_strategy_sv.value());
+        } catch (const std::runtime_error& e) {
```

Comment:
Strange that the unused param is not an error.

Reply:
I thought the same thing!